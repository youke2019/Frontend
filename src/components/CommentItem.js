import React from "react";
import {
  View,
  Text,
  Image,
  StyleSheet,
  TouchableOpacity
} from "react-native"
import ReplyBox from './ReplyBox'
import { UserIdText } from './UserIdText'
import { praiseComment, sendCommentReply, unPraiseComment } from '../utils/DataRequest'
import { UserAvatarImg } from './UserAvatarImg'



 class CommentItem extends React.Component {
   constructor (props){
     super(props)
     const liked = props.comment_info != null ? props.comment_info.current_user_praise : false;
    this.state = {
      reply_visible: false,
      liked: liked,
    }
   }
  componentWillReceiveProps (nextProps, nextContext) {
    const liked = nextProps.comment_info != null ? nextProps.comment_info.current_user_praise : false;
    this.setState({
      liked:liked,
    })
  }

   onPressComment = ()=>{
    this.setState({
      reply_visible:true,
    })
  }
  onReplyDone = (msg)=>{
    sendCommentReply(this.props.comment_info.course_comment_id,this.props.user_info.id,msg)
      .then(response=>{
        this.props.refresh();
      }).catch(err=>console.log(err))
    this.closeComment()
  }
  closeComment =()=>{
    this.setState({
      reply_visible:false,
    })
  }
  onPressLike = () =>{
    const {user_info,comment_info} = this.props
    if(!this.state.liked) {
      praiseComment({
        user_id: user_info.id,
        course_comment_id: comment_info.course_comment_id,
      }).then((response) => {
        console.log("resfresh")
        this.props.refresh();
      }).catch((err) => {console.log(err);})
    }else {
      unPraiseComment({
        user_id: user_info.id,
        course_comment_id: comment_info.course_comment_id,
      }).then(() => {
        this.props.refresh();
      }).catch((err) => {console.log(err);})
    }

  }
  render(){
    const {
      comment_info = null,
    } = this.props;

    const {reply_visible} = this.state;
    return (
      <View >
      {
        comment_info === null ?
          (<View style = {styles.empty_container}>
            <View
              style ={{width:340,height:100,flexDirection:'column',alignItems:'center',justifyContent:'center'}}
            >
              <Text style={styles.empty_msg}> ËøòÊ≤°Êúâ‰∫∫ËØÑËÆ∫ÔºåÂø´Êù•Êä¢Âç†Ê≤ôÂèëÂêßüòã</Text>
            </View>
          </View>)
          :
          (<View
              style={styles.card_container}
            >
            <View style = {styles.container}>
              <View style ={styles.comment}>
                <View style={styles.comment_header}>
                  <UserAvatarImg
                    style={styles.avatar}
                    img_style={styles.avatar_img}
                    user_id={comment_info.user_id}
                  />
                  <UserIdText style = {styles.user_id} user_id={comment_info.user_id}/>
                </View>
                <View style={styles.comment_body_container}>
                  <Text style = {styles.comment_body}> {comment_info.course_comment_content}</Text>
                </View>
                <View style={styles.comment_bottom}>
                  <Text style={{fontSize: 12, padding: 5}}>{comment_info.course_comment_time}</Text>
                  <TouchableOpacity
                      onPress={this.onPressComment}
                      style = {styles.button_comment}
                      activeOpacity={0.3}
                  >
                    <Image source={{uri:'pinglun'}} style={{width:14,height:14,resizeMode: 'contain'}}/>
                    <Text style={{fontSize: 12,padding: 5,}}>ËØÑËÆ∫</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                      onPress={this.onPressLike}
                      style = {styles.button_like}
                      activeOpacity={0.3}
                  >
                    <Image source={{uri:'dianzan'}} style={ this.state.liked ?   {width:14,height:14,resizeMode: 'contain',tintColor:'orange'}: {width:14,height:14,resizeMode: 'contain'}}/>
                    <Text style={{fontSize: 12,padding: 5}}>{comment_info.course_comment_praise_point }</Text>
                  </TouchableOpacity>
                </View>
                <View style={styles.reply_area}>
                  {
                    comment_info.courseCommentReplyList.map((item, index) => {
                      return  (
                          <View style={styles.reply_item} key={index}>
                            <UserIdText user_id={item.user_id} style={styles.reply_user_id}/>
                            <Text style={styles.reply_text}>{': ' + item.course_comment_reply_content}</Text>
                          </View>)
                    })
                  }
                </View>
              </View>
            </View>
          </View>)
      }
      <ReplyBox
        visible = {reply_visible}
        onReplyDone = {this.onReplyDone}
        onBackdropPress={this.closeComment}
      />
      </View>
    )
  }
}
const styles = StyleSheet.create({
  card_container:{
    paddingHorizontal:10,
    paddingVertical: 10,
    alignItems: 'flex-start',
    borderRadius: 20,
    elevation: 4,
    backgroundColor: '#FFFFFF',
  },
  container:{
    flexDirection:'row',
  },
  avatar:{
    flexDirection: 'column',
    justifyContent:'flex-start',
  },
  avatar_img:{
    width:40,
    height:40,
    borderRadius:20,
    marginRight: 8,
  },
  empty_container:{
    flexDirection:'column',
    alignItems: 'center',
  },
  empty_msg:{
    paddingVertical:20,
    alignItems: 'center',
  },
  comment:{
    flex:1,
    marginLeft:10,
    marginRight:10,
  },

  comment_header:{
    flexDirection:'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingTop:12,
  },
  user_id:{
    fontSize: 15,
    color: '#000000',
  },
  comment_body_container:{
    paddingVertical: 12,
  },
  comment_body:{
    paddingVertical:20
  },
  comment_bottom:{
    flex:1,
    flexDirection:'row',
    justifyContent: "space-around",
    alignItems:'center',
    paddingBottom: 10,
  },
  button_like:{
    height:20,
    flexDirection:'row',
    alignItems:'center'
  },
  button_comment:{
    flexDirection:'row',
    alignItems:'center',
    height:20,
  },
  reply_area: {
    borderRadius:5,
    paddingBottom: 20,
  },
  reply_item: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 3,
  },
  reply_user_id: {
    fontSize: 14,
    fontWeight: 'bold'
  },
  reply_text: {
    fontSize: 14,
    lineHeight: 18,
  }
});

export default CommentItem