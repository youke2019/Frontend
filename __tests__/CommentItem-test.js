import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16/build'
import CommentItem from '../src/components/CommentItem'

configure({adapter: new Adapter()})

jest.mock("../src/utils/DataRequest.js")
const comment_info = JSON.parse("{\"course_comment_id\":14,\"course_id\":\"BM176\",\"course_comment_time\":\"2019-07-18 23:07:37\",\"course_comment_content\":\"我觉得这门课很垃圾\",\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"isbanned\":false,\"course_comment_praise_point\":1,\"current_user_praise\":true,\"courseCommentReportList\":[],\"courseCommentReplyList\":[{\"course_comment_reply_id\":1,\"course_comment_reply_content\":\"服服服\",\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"course_comment_id\":14},{\"course_comment_reply_id\":2,\"course_comment_reply_content\":\"八九十年代不\",\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"course_comment_id\":14},{\"course_comment_reply_id\":3,\"course_comment_reply_content\":\"同意呀\",\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"course_comment_id\":14}]}")
const user_info = {
  id:'test_id'
}
test('renders CommentAbstract component correctly', (done) => {
  const tree = renderer.create(
      <CommentItem
        comment_info={comment_info}
        user_info={user_info}
        refresh={()=>{}}
      />
  ).toJSON();
  setTimeout(()=>{
    expect(tree).toMatchSnapshot();
    done();
  },1000)
});
test('renders CommentAbstract component correctly', (done) => {
  const null_tree = renderer.create(
    <CommentItem
      user_info={user_info}
      refresh={()=>{}}
    />
  ).toJSON();
  setTimeout(()=>{
    expect(null_tree).toMatchSnapshot();
    done();
  },1000)
});
describe("<CommentItem/> component",()=>{
  const wrapper = shallow(
    <CommentItem
  comment_info={comment_info}
  user_info={user_info}
  refresh={()=>{}}
  />
  );
  const instance = wrapper.instance();
  console.log(instance)
  it("react when press like",(done)=>{
    instance.setState({
      liked:true
    })
    instance.onPressLike();
    instance.onPressLike();
    instance.setState({
      liked:false
    })
    instance.onPressLike();
    instance.onPressLike();
    setTimeout(()=>{done();},1000) /*wait for async to end*/
  })
  it("send msg when Reply Done",(done)=>{
    instance.onReplyDone()
    instance.onReplyDone()
    setTimeout(()=>{done();},1000) /*wait for async to end*/
  })
  it("open comment when press Comment",()=>{
    instance.onPressComment()
    setTimeout(()=>{done();},1000) /*wait for async to end*/
  })
  it("refresh when receive props",()=>{
    instance.componentWillReceiveProps({comment_info:{current_user_praise:true}},null)
    instance.componentWillReceiveProps({comment_info:null},null)
    setTimeout(()=>{done();},1000) /*wait for async to end*/
  })
})