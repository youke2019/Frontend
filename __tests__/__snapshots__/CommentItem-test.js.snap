// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommentAbstract component correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 20,
        "elevation": 4,
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingTop": 12,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "default_avatar_1",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": 40,
                  "marginRight": 8,
                  "width": 40,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 15,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
             
            
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingVertical": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "paddingVertical": 20,
              }
            }
          >
             
            我觉得这门课很垃圾
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-around",
              "paddingBottom": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
                "padding": 5,
              }
            }
          >
            2019-07-18 23:07:37
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 20,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "pinglun",
                }
              }
              style={
                Object {
                  "height": 14,
                  "resizeMode": "contain",
                  "width": 14,
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 12,
                  "padding": 5,
                }
              }
            >
              评论
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 20,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "dianzan",
                }
              }
              style={
                Object {
                  "height": 14,
                  "resizeMode": "contain",
                  "tintColor": "orange",
                  "width": 14,
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 12,
                  "padding": 5,
                }
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 5,
              "paddingBottom": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 3,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
               
              
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              : 服服服
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 3,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
               
              
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              : 八九十年代不
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 3,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "bold",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
               
              
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              : 同意呀
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "borderColor": "lightgrey",
          "borderTopWidth": 0.2,
          "bottom": 0,
          "height": 50,
          "paddingHorizontal": 10,
          "paddingVertical": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        keyboardShouldPersistTaps="always"
        style={
          Object {
            "contentSize": 0,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "whitesmoke",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "orange",
                  "borderRadius": 8,
                  "borderWidth": 0.5,
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                autoFocus={true}
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                placeholder="点击输入回复"
                rejectResponderTermination={true}
                style={
                  Object {
                    "fontSize": 15,
                    "padding": 2,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1.5,
                  "marginVertical": 0,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FDD326",
                    "borderRadius": 8,
                    "height": "auto",
                    "marginHorizontal": 8,
                    "paddingVertical": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "字魂70号-灵悦黑体",
                      "fontSize": 15,
                      "lineHeight": 15,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                   发送
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </Modal>
</View>
`;

exports[`renders CommentAbstract component correctly 2`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": 100,
          "justifyContent": "center",
          "width": 340,
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "paddingVertical": 20,
          }
        }
      >
         还没有人评论，快来抢占沙发吧😋
      </Text>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "borderColor": "lightgrey",
          "borderTopWidth": 0.2,
          "bottom": 0,
          "height": 50,
          "paddingHorizontal": 10,
          "paddingVertical": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        keyboardShouldPersistTaps="always"
        style={
          Object {
            "contentSize": 0,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "whitesmoke",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "orange",
                  "borderRadius": 8,
                  "borderWidth": 0.5,
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginVertical": 10,
                  "paddingHorizontal": 10,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                autoFocus={true}
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                placeholder="点击输入回复"
                rejectResponderTermination={true}
                style={
                  Object {
                    "fontSize": 15,
                    "padding": 2,
                    "textAlign": "left",
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1.5,
                  "marginVertical": 0,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FDD326",
                    "borderRadius": 8,
                    "height": "auto",
                    "marginHorizontal": 8,
                    "paddingVertical": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "字魂70号-灵悦黑体",
                      "fontSize": 15,
                      "lineHeight": 15,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                   发送
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </Modal>
</View>
`;
